### PR TITLE
Improve Icon color cascade when using SVGSource

### PR DESCRIPTION
**Pull request by gil--**
When using an Icon source of type SVGSource, the SVG may have inline fill which should be overwritten based on the Icon React color prop value.

### WHY are these changes introduced?

When using an Icon `source={SVGSource}`, if the SVGSource contains inline fill, it overrides the Icon color prop. Imo, the React Icon component should override any fill on the Icon being included if you have a color prop.

### WHAT is this pull request doing?

Overrides any inline fill on an SVG with the Icon color prop value.

screen shot 2019-01-09 at 10 44 44 pm

**Inline SVG fill is overriding the Icon color prop:**
screen shot 2019-01-09 at 10 44 32 pm

**With wildcard cascade:**
screen shot 2019-01-09 at 10 44 21 pm
screen shot 2019-01-09 at 10 49 43 pm

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris-react/blob/master/README.md#development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md)

```jsx

import React from 'react'
import styled from 'styled-components'
import { Icon, TextStyle } from '@shopify/polaris'

const orderIcon = (
    <svg xmlns="http://www.w3.org/2000/svg" width="16" height="16"><g className="nc-icon-wrapper" fill="#444"><path d="M15 3H4.5L4 .8C3.9.3 3.5 0 3 0H0v2h2.2L4 10.2c.1.5.5.8 1 .8h8c.4 0 .8-.3.9-.7l2-6c.2-.5-.1-1.3-.9-1.3z"/><circle data-color="color-2" cx="5" cy="14" r="2"/><circle data-color="color-2" cx="13" cy="14" r="2"/></g></svg>
)

const OrderTotal = ({ total }) => (
    <Total>
        <Icon source={orderIcon} color="inkLightest" />
        <TextStyle variation="strong">{total}</TextStyle>
    </Total>
)

export default OrderTotal

```

### 🎩 checklist

* [X] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
* [X] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
* [X] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)

### CODE REVIEW

File: UNRELEASED.md [hunk 1]
```diff
@@ -6,6 +6,8 @@ Use [the changelog guidelines](https://git.io/polaris-changelog-guidelines) to f
 
 ### Enhancements
 
+- Icon component color prop now cascades to child SVGs when using `SVGSource` as **source** prop.
```
Comment (kaelig) [inline UNRELEASED.md]: Should "child" be plural?

Comment (kaelig) [reply UNRELEASED.md]: Thanks! In French we'd pluralize both the adjective and the noun – hence my question!

Comment (gil--) [reply UNRELEASED.md]: @kaelig  Happy to change if you think it's needed. English is hard. 😄

Comment (kaelig) [reply UNRELEASED.md]: All good – I'll follow @dpersing's guidance on this!

**Comment by dpersing** [comment]
Updated `UNRELEASED.md`. I'll check Percy once checks are done again and then this should be good to squash and merge.

**Comment by danrosenthal** [comment]
@gil-- I am going to go ahead and close this for now, as there hasn't been much activity on it lately. If this is still a change you'd like to see, please open it back up, get it up to date, and ping a couple of us for review!

**Pull request closed without merging**